enhancements/20220822-storage-network/20220822-storage-network.md

```diff
@@ -0,0 +1,163 @@
+
+VM and Storage Network will use different interfaces.
+
+- VM uses eth2 and eth3
+- Storage uses eth4 and eth5
+- Create a new VlanConfig and ClusterNetwork with eth4 and eth5
+
+#### Story 3
+
+Users want to rollback the config back to default
+
+### User Experience In Detail
+
+#### Controller
+
+1. Users have to shutdown all VMs and stop all VMI before setting `storage-network` in Harvester settings
+2. Users need to create a VLAN Config with ClusterNetwork name and ensure the cluster network and VLAN config cover all nodes and network connectivity is expected.
+3. Users have to ensure the Cluster Network is configured and VLAN Config will cover all nodes and ensure the network connectivity is working and expected in all nodes.
+4. Users need to assign ClusterNetwork, VLAN ID, IP Range in storage-network settings
+5. Controller will shut down management pods attached to Longhorn volumes and check all volumes are detached.
+6. Controller will set `storage-network` config to Longhorn settings `storage-network`.
+7. Controller will restart management pods.
+8. Users need to restart VM manually.
+
+### API changes
+
+- New Settings `storage-network`
+
+## Design
+
+### Implementation Overview
+
+- Re-use VLAN Bridge from VLAN enhancement
+- Introduce IPAM [Whereabouts](https://github.com/k8snetworkplumbingwg/whereabouts) to be a global IPAM for Longhorn Pods.
+- Create a Network Attachment Definition with plugin bridge and Whereabouts IPAM
+- According to [Longhorn storage network limitation](https://longhorn.io/docs/1.3.1/advanced-resources/deploy/storage-network/), users should not modify NAD after setting in Longhorn settings.
+- Controller will create NAD for storage-network, and remove old NAD.
+- Controller will sync storage-network config to Longhorn storage-network setting
+
+### Controller Flow Chart
+
+![flow chart](./storagenetwork.png)
+
+### Resource needs to be paused or stopped
+
+- Rancher Monitoring
+    - kind: managedchart
+    - namespace: fleet-local
+    - name: rancher-monitoring
+- Grafana
+    - kind: deployments
+    - namespace: cattle-monitoring-system
+    - name: rancher-monitoring-grafana
+- Prometheus
+    - kind: prometheus
+    - namespace: cattle-monitoring-system
+    - name: rancher-monitoring-prometheus 
+- Alertmanager
+    - kind: alertmanager
+    - namespace: cattle-monitoring-system
+    - name: rancher-monitoring-alertmanager
+- Harvester VM Import Controller
+    - kind: deployments
+    - namespace: harvester-system
+    - name: harvester-vm-import-controller
+
+#### Settings config example
+```json
+{
+    "vlan": 100,
+    "clusterNetwork": "mgmt",
+    "range": "192.168.0.0/24"
+}
+```
+
+#### NAD config example
+```json
+{
+    "cniVersion":"0.3.1",
+    "name":"v1",
+    "type":"bridge",
+    "bridge":"BRIDGE_NAME",
+    "vlan":VLAN_ID,
+    "ipam":{
+        "type":"whereabouts",
+        "range":"192.168.0.0/24"
+    }
+}
+```
+
+### Test plan
+
+#### apply network
+
+1. Shutdown all VMs
```

Comment:
```suggestion
1. Shutdown all VMs.
```
And other lines. Please check if all lists in the document end with a period. We need to be consistent.